app.rb

```diff
@@ -0,0 +1,172 @@
+require 'date'
+
+require './book'
+require './student'
+require './rental'
+require './teacher'
+require './user_input'
+
+class App
+  def initialize(main_choices, books = [], people = [])
+    @books = books
+    @people = people
+    @main_choices = main_choices
+  end
+
+  def run
+    loop do
+      puts @main_choices
+      choice = UserInput.get_number('', 'Please enter a valid number!')
+      return if choice == 7
+
+      run_available_options(choice)
+    end
+  end
+
+  private
+
+  def run_available_options(choice)
+    case choice
+    when 1
+      list_books(@books)
+    when 2
+      list_people(@people)
+    when 3
+      add_person
+    when 4
+      @books << add_book
+    when 5
+      add_rental(@books, @people)
+    when 6
+      list_rentals(@people)
+    else
+      puts ''
+    end
+  end
+
+  def add_person
+    person_type = UserInput.get_number(
+      'Do you want to create a Student (1) or a teacher (2)?. Back to main (any number): ',
+      'Please enter a valid number!'
+    )
+
+    case person_type
+    when 1
+      @people << add_student
+    when 2
+      @people << add_teacher
+    else
+      puts ''
+    end
+  end
+
+  def add_student
+    age = UserInput.get_number('Age: ', 'Please enter a valid number!')
+    name = UserInput.get_str('Name: ', 'Please enter a proper name!')
+    permission = UserInput.get_bool('Parent permission (y/n): ', 'Please enter either (y/n)!')
+    student = Student.new(age, name, parent_permission: permission)
+    return unless student
+
+    puts "Student #{name} is added!\n\n"
+    student
+  end
+
+  def add_teacher
+    specialization = UserInput.get_str('Specialization: ', 'Please enter a proper specialization!')
+    age = UserInput.get_number('Age: ', 'Please enter a valid number!')
+    name = UserInput.get_str('Name: ', 'Please enter a proper name!')
+    teacher = Teacher.new(specialization, age, name)
+    return unless teacher
```

Comment:
- [ ] Please notice that an `ArgumentError` occurs when attempting to create a new `Teacher`. Kindly see the error below and try to fix it 👇 
![image](https://user-images.githubusercontent.com/46839548/231997785-880e1d2c-8854-4f46-821c-86960c04dd6a.png)